animation_nodes/nodes/object/object_viewport_color.py

```diff
@@ -0,0 +1,30 @@
+import bpy
+from ... base_types import AnimationNode, VectorizedSocket
+
+class ObjectViewportColorNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ObjectViewportColorNode"
+    bl_label = "Object Viewport Color"
+    codeEffects = [VectorizedSocket.CodeEffect]
+
+    useObjectList: VectorizedSocket.newProperty()
+    useColorList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Objects", "objects")))
```

Comment:
The object input should have `codeProperties = dict(allowListExtension = False)`. `allowListExtension` is the property that tells Animation Nodes to repeat this list if the other lists were larger (Similar to virtual lists). It is True by default, and we don't want objects to repeat.

Reply:
Okay.

Reply:
Omar, I have added this code,
```
self.newInput(VectorizedSocket("Object", "useObjectList",
            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
            ("Objects", "objects")), codeProperties = dict(allowListExtension = False))
```
When I input color list larger than the input objects, it slows down the node-tree.

Reply:
This should be:
```python
self.newInput(VectorizedSocket("Object", "useObjectList",
            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
            ("Objects", "objects"),
            codeProperties = dict(allowListExtension = False)))
```

Reply:
Done.